app/forms/FastPassForm.scala

```diff
@@ -0,0 +1,191 @@
+    def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[String]] = {
+      implicit val keyQualifier = getKeyQualifier(key)
+      bindOptionalParam(data.isApplicable, data.isValidFastPassTypeSelected,
+        fastPassTypeRequiredMsg)(key, data.getFastPassType)
+    }
+
+    def unbind(key: String, value: Option[String]): Map[String, String] = optionalParamToMap(key, value)
+
+  }
+
+  def internshipTypesFormatter = new Formatter[Option[Seq[String]]] {
+    def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[Seq[String]]] = {
+      implicit val keyQualifier = getKeyQualifier(key)
+      bindOptionalParam(data.isDiversityInternshipSelected, data.isValidInternshipTypesSelected,
+        internshipTypeRequiredMsg)(key, data.getInternshipTypes)
+    }
+
+    def unbind(key: String, value: Option[Seq[String]]): Map[String, String] = {
+      value match {
+        case Some(seq) => seq.zipWithIndex.foldLeft(Map.empty[String, String])(
+          (res, pair) => res + (s"$key[${pair._2}]" -> pair._1))
+        case None => Map.empty[String, String]
+      }
+    }
+  }
+
+  def fastPassReceivedFormatter = new Formatter[Option[Boolean]] {
+    def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[Boolean]] = {
+      implicit val keyQualifier = getKeyQualifier(key)
+      bindOptionalParam(data.isSDIPCurrentYearSelected, data.isFastPassReceivedValid,
+        fastPassReceivedRequiredMsg)(key, data.getFastPassReceived.toBoolean)
+    }
+
+    def unbind(key: String, value: Option[Boolean]): Map[String, String] = optionalParamToMap(key, value)
+
+  }
+
+  def fastPassCertificateFormatter = new Formatter[Option[String]] {
+    def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[String]] = {
+      implicit val keyQualifier = getKeyQualifier(key)
+      bindOptionalParam(data.isFastPassReceived, data.isCertificateNumberValid,
+        certificateNumberRequiredMsg) (key, data.getCertificateNumber)
+    }
+
+    def unbind(key: String, value: Option[String]): Map[String, String] = optionalParamToMap(key, value)
+  }
+
+  private def bindOptionalParam[T](dependencyCheck: Boolean, validityCheck: Boolean, errMsg: String)
+                                  (key: String, value: => T):Either[Seq[FormError], Option[T]] =
+    (dependencyCheck, validityCheck) match {
+      case (true, false) => Left(List(FormError(key, errMsg)))
+      case (true, true) => Right(Some(value))
+      case (false, _) => Right(None)
+    }
+
+
+  private def optionalParamToMap[T](key: String, optValue: Option[T]) = {
+    optValue match {
+      case None => Map.empty[String, String]
+      case Some(value) => Map(key -> value.toString)
+    }
+  }
+
+  implicit class DataValidations(data: Map[String, String]) {
+
+    def getApplicable(implicit keyQualifier: String) = data.getOrElse(s"$keyQualifier$applicable", "")
```

Comment:
It can't be necessary for this to be this complex, lets talk about doing that case class thing as a follow on pull request, bearing in mind Lukasz is waiting to build on this one. I had to open this in an IDE to follow it around